File: 008_Array_Part_1/MinMax.py
import sys

arr = [20, -47, 52, 59, -12]
size = len(arr)

# Find the smallest element
smaller = sys.maxsize
for i in range(size):
    smaller = min(arr[i], smaller)
print(smaller)

# Find the largest element and its index
larger = -sys.maxsize - 1
index = -1
for i in range(size):
    if arr[i] > larger:
        larger = arr[i]
        index = i
print(larger)
print(index)

File: 008_Array_Part_1/MinMax.cpp
#include<iostream>
#include <climits>
using namespace std;

int main(){
    int arr[5] = {20, -47, 52, 59, -12};
    int size = 5;
    int smaller = INT_MAX;
    int index = -1;
    for(int i=0; i<size; i++){
        // if (arr[i] < smaller)
        // smaller = arr[i];
        smaller = min(arr[i], smaller);
    }
    cout<<smaller<<endl;

    int larger = INT_MIN;
    for(int i=0; i<size; i++){
        if (arr[i] > larger){
            larger = arr[i];
            index = i;
        }
        
        // larger = max(larger, arr[i]);
    }
    cout<<larger<<endl;
    cout<<index;

    return 0;
}